packages/origin.js/test/resource_purchases.test.js

```diff
@@ -56,45 +59,69 @@ describe("Purchase Resource", function() {
     purchase = await purchases.get(purchaseEvent.args._purchaseContract)
   }
 
+  let expectStage = function(expectedStage) {
+    expect(purchase.stage).to.equal(expectedStage)
+  }
+
+  // Tests
+  // -----
+
   describe("simple purchase flow", async () => {
     before(async () => {
       await resetListingAndPurchase()
     })
 
     it("should get a purchase", async () => {
-      expect(purchase.stage.toNumber()).to.equal(0)
+      expectStage(purchases.STAGES.AWAITING_PAYMENT)
```

Comment:
Corresponding with my other comment, I don't think we should be accessing our constant definitions from the class that we're testing.

1. It's not good design to export something just so it can be used in a test
2. Our tests should not be retrieving values to expect from the very code they are testing. If we make a mistake in one of those definitions, the tests will not catch it because they are using the exact same definitions.

So although it is a bit redundant, I believe it's best to simply define the stages as constants at the top of our test file. (We're already doing this in the contract tests.) At the cost of having to do a bit more typing when changing values, it ensures that our tests will not be fooled by mistakes in our source. If someone updates a constant in a source file but forgets to update the tests, worst that happens is that tests fail. 😃

Reply:
I think I see what you are saying here - we want to make sure that what we "promise" in the API in terms of string stage names  is tested so that we don't accidentally change it, and break downstream code.

I've added a section to the tests that checks that these do not change.

The stage-name-strings-as a constant isn't just intended for our tests - it's for downstream code to use too. It's a pretty common javascript idiom for enum-like strings to be accessed through namespaced constants like this.